test/functional/aws_legacy_cluster_vpc.go

```diff
@@ -0,0 +1,284 @@
+	}
+
+	// create a new session
+	session, err := CreateAWSSession(ctx, testingCtx.Client)
+	if err != nil {
+		t.Fatal("could not create aws session", err)
+	}
+
+	ec2Svc := ec2.New(session)
+
+	// Get the availability zones for the region the cluster is in
+	azs, err := getAvailabilityZones(ec2Svc)
+
+	if err != nil {
+		t.Fatal("could get aws availability zones", err)
+	}
+
+	// find and verify the cluster VPC
+	vpc, err := verifyLegacyVPC(ec2Svc, clusterTag)
+	testErrors.vpcError = err.(*networkConfigTestError).vpcError
+
+	clusterVpcId := *vpc.VpcId
+
+	vpcSubnets, err := verifyLegacySubnets(ec2Svc, clusterTag, clusterVpcId)
+	testErrors.subnetsError = err.(*networkConfigTestError).subnetsError
+
+	// we have to manually construct the subnet group names for rds and elasticache,
+	// since tag filtering isnt currently available
+	subnetGroupName := resources.ShortenString(fmt.Sprintf("%s-%s", clusterTag, "subnet-group"), 40)
+
+	rdsSvc := rds.New(session)
+	err = verifyLegacyRDSSubnetGroups(rdsSvc, subnetGroupName, clusterVpcId, vpcSubnets, azs)
+	testErrors.rdsSubnetGroupsError = err.(*networkConfigTestError).rdsSubnetGroupsError
+
+	cacheSvc := elasticache.New(session)
+	err = verifyLegacyCacheSubnetGroups(cacheSvc, subnetGroupName, clusterVpcId, vpcSubnets, azs)
+	testErrors.cacheSubnetGroupsError = err.(*networkConfigTestError).cacheSubnetGroupsError
+
+	// if any error was found, fail the test
+	if testErrors.hasError() {
+		t.Fatal(testErrors.Error())
+	}
+}
+
+func getAvailabilityZones(ec2Svc *ec2.EC2) ([]*string, error) {
+	azs := make([]*string, 0)
+
+	describeOutput, err := ec2Svc.DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{})
+
+	if err != nil {
+		return azs, err
+	}
+
+	for _, zone := range describeOutput.AvailabilityZones {
+		azs = append(azs, zone.ZoneName)
+	}
+
+	return azs, nil
+}
+
+func verifyLegacyVPC(session *ec2.EC2, clusterTag string) (*ec2.Vpc, error) {
+	// filter vpcs by integreatly cluster id tag
+
+	newErr := &networkConfigTestError{
+		vpcError: []error{},
+	}
+
+	describeVpcs, err := session.DescribeVpcs(&ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String(fmt.Sprintf("tag:kubernetes.io/cluster/%s", clusterTag)),
+				Values: []*string{aws.String("owned")},
+			},
+		},
+	})
+	if err != nil {
+		newErr.vpcError = append(newErr.vpcError, fmt.Errorf("could not find vpc: %v", err))
+		return nil, newErr
+	}
+
+	// only one vpc is expected
+	vpcs := describeVpcs.Vpcs
+	if len(vpcs) != 1 {
+		newErr.vpcError = append(newErr.vpcError, fmt.Errorf("expected 1 vpc but found %d", len(vpcs)))
+		return nil, newErr
+	}
+
+	// cidr blocks should match
+	return vpcs[0], newErr
+}
+
+// verify that the vpc subnets are created
+func verifyLegacySubnets(session *ec2.EC2, clusterTag, clusterVPCId string) ([]*ec2.Subnet, error) {
+	newErr := &networkConfigTestError{
+		subnetsError: []error{},
+	}
+
+	// filter subnets by integreatly cluster id tag
```

Comment:
Are you using the `vpc-id` tag instead now?

Reply:
Would it also be worth checking how many subnets we expect?

Reply:
Yeah I'm using the `vpc-id` tag because a lot of the subnets are created by the openshift installer and don't have the `integreatly.org/clusterID` tag on them.

It may be worth checking the subnets but I'm not too sure seeing as it's a little outside of our control.

Reply:
+1 it is not as deterministic as the standalone network, as we basically ensure there is a private subnet for every az. I think if we add that check to the subnet groups, we should be okay there

Reply:
Sure, makes sense